src/mscorlib/shared/System/HashCode.cs

```diff
@@ -0,0 +1,433 @@
+            // _queue2 and _queue3 are basically a buffer so that when ToHashCode is
+            // called we can execute *2 correctly. That's what first three case
+            // statements do.
+
+            // We need to initialize the xxHash32 state (_v1 -> _v4) lazily (see *0)
+            // and the last place that can be done if you look at the original code
+            // is just before the first block of 16 bytes is mixed in. The xxHash32
+            // state is never used for streams containing fewer than 16 bytes.
+
+            // A bloom filter is used to determine whether the default case statement
+            // has even been executed. To do that we check if the length is smaller
+            // than 4 (_length ^ position will be non-zero if this is the case). The
+            // case statement is for values larger than 2, so the check only succeeds
+            // on exactly 3.
+
+            // To see what's really going on here, have a look at the Combine methods.
+            
+            var val = (uint)value;
+            uint position = _length % 4;
+
+            // Switch can't be inlined.
+
+            if (position == 0)
+                _queue1 = val;
+            else if (position == 1)
+                _queue2 = val;
+            else if (position == 2)
+                _queue3 = val;
+            else // == 3
+            {
+                // length smaller than 4?
+                if ((_length ^ position) == 0)
+                    Initialize(out _v1, out _v2, out _v3, out _v4);
+
+                _v1 = Round(_v1, _queue1);
+                _v2 = Round(_v2, _queue2);
+                _v3 = Round(_v3, _queue3);
+                _v4 = Round(_v4, val);
+            }
+
+            // Throw for more than uint.MaxValue fields.
+            _length = checked(_length + 1);
+        }
+
+        public int ToHashCode()
+        {
+            uint position = _length % 4;
+
+            // If the length is less than 3, _v1 -> _v4 don't contain
+            // anything yet. xxHash32 treats this differently.
+
+            uint hash = (_length ^ position) == 0
+                ? MixEmptyState()
+                : MixState(_v1, _v2, _v3, _v4);
+
+            // Multiply by 4 because we've been counting in ints, not
+            // bytes.
+
+            hash += _length * 4;
+
+            // Mix what remains in the queue
+            // Switch can't be inlined right now, so use as few
+            // branches as possible instead.
+
+            if (position > 0)
+            {
+                hash = QueueRound(hash, _queue1);
+                if (position > 1)
+                {
+                    hash = QueueRound(hash, _queue2);
+                    if (position > 2)
+                        hash = QueueRound(hash, _queue3);
+                }
+            }
+
+            hash = MixFinal(hash);
+            return (int)hash;
+        }
+
+#       pragma warning disable 0809
```

Comment:
Nit: unindent the pragma per .NET Core coding conventions. (same for below pragma)